department-of-veterans-affairs/caseflow
# Converted judge assignment

Resolves [CASEFLOW-1618](https://vajira.max.gov/browse/CASEFLOW-1618)

**PR Stack**

[Part 1](https://github.com/department-of-veterans-affairs/caseflow/pull/16577)
Part 2
[Part 3](https://github.com/department-of-veterans-affairs/caseflow/pull/16579)
[Part 4](https://github.com/department-of-veterans-affairs/caseflow/pull/16580)

### Description
Changes the Judge Assignment algorithm to directly assign Judges as opposed to assigning them through an algorithm

### Acceptance Criteria
- [ ] Add hearing day ID and judge ID columns to hearing schedule CSV download (but not Caseflow UI)
- [ ] A user with "Build HearSched" permissions can upload of that CSV/spreadsheet at /hearings/schedule/build/
  - [ ] User still selects "Judge non-availability" radio button...
  - [ ] but radio button label now reads "Judge assignment" instead of "Judge non-availability"
  - [ ] User does not need to select start date or end date
  - [ ] Validate that judge IDs and names match on upload (current behavior)
- [ ] Upon upload and validation, the user sees a preview of the total # of hearing days that will be updated and which ones before confirming the assignment
- [ ] When the user confirms, the specified judges are assigned to each to each hearing day
- [ ] Users can repeat the upload and assignment for a given date range so they can leave some days unassigned or reassign days as necessary
- [ ] Judge assignment template is removed as an option from the Download templates option dropdown on /hearings/schedule/build
- [ ] Figure out and fix the blank Type column in CSV

### Testing Plan
1. Shadow user `BVASYELLOW` http://localhost:3000/test/users
2. Navigate to the hearings schedule http://localhost:3000/hearings/schedule
3. Click the "Download Current View" button and open the CSV
4. Remove all but the first 10 rows or so, shuffle the Judge ID/Name pairs so they are associated with different hearing days and save the document as an Excel doc
5. Click the "Build Schedule" button and then click the "Upload Files" button on the next page
6. Choose the "Judge Assignment" option and select the file you just uploaded
7. Click the continue button and verify the assignments in the table on the next page
8. Click the go back button and ensure that you are taken to the previous page without the changes being saved
9. Click continue again and this time click the "Confirm Assignments" button and ensure you are taken back to the hearings schedule with a success banner and the updated assignments

### User Facing Changes

 BEFORE|AFTER
 ---|---
